server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java

```diff
@@ -159,35 +148,49 @@ public void sendingTenMailsShouldBeRandomlyAssignedToEveryUsersINBOX() throws Ex
                     .sendMessage(buildMail("Message " + index))).sneakyThrow());
 
         Collection<IMAPMessageReader> connexions = USERS
-                .stream()
-                .map(Throwing.function(username -> new IMAPMessageReader()
-                    .connect(LOCALHOST_IP, imapProbe.getImapPort())
-                    .login(username, PASSWORD)))
-                .collect(Guavate.toImmutableList());
+            .stream()
+            .map(this::createIMAPConnexion)
+            .collect(Guavate.toImmutableList());
 
         awaitAtMostTenSeconds
-            .untilAsserted(() -> checkMailboxesHaveBeenFilled(connexions));
+            .untilAsserted(() -> checkMailboxesHaveBeenFilled(connexions, numberOfMails));
     }
 
-    private void checkMailboxesHaveBeenFilled(Collection<IMAPMessageReader> connexions) {
+    private IMAPMessageReader createIMAPConnexion(String username) {
+        try {
+            return new IMAPMessageReader()
+                .connect(LOCALHOST_IP, imapProbe.getImapPort())
+                .login(username, PASSWORD);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void checkMailboxesHaveBeenFilled(Collection<IMAPMessageReader> connexions, int numberOfMails) {
         assertThat(connexions
             .stream()
-            .mapToLong(this::mailboxIsFilled)
+            .flatMapToLong(this::numberOfAnUserMessages)
```

Comment:
it's a user

https://ell.stackexchange.com/questions/26986/why-a-user-instead-of-an-user that unicorn 2000 forum will summarize it well...